### PR TITLE
Fix torch backend cross() using wrong default axis

### PR DESCRIPTION
**Pull request by CodingSelim**
- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

---

on the torch backend ops.cross passed dim=axis directly to torch.cross. when axis is None (the default) torch.cross falls back to the first dimension of size 3, but numpy and the numpy/jax/tf backends use the last axis. so the torch result was wrong whenever an earlier axis was also length 3, and it printed a deprecation warning too.

small repro:

```python
import numpy as np, keras
a = np.arange(9.).reshape(3, 3); b = np.arange(9.)[::-1].reshape(3, 3)
keras.ops.cross(a, b)   # torch uses axis 0, doesnt match np.cross(a, b)
```

fix is to default axis to -1 before the call, same as the other backends. also switched to torch.linalg.cross so the deprecated one is gone.

added a (2,3,3) case to test_cross that fails on the old code. the old test used ones([2,1,4,3]) where only the last axis is size 3 and all ones cross to zero so it never caught this.

### CODE REVIEW

**Comment by CodingSelim** [comment]
Good call, switched to torch.linalg.cross. Verified the regression test and 1D case still pass with no deprecation warning.

**Comment by CodingSelim** [comment]
@googlebot I signed it!

**Review by hertschuh — APPROVED** [review]
Thanks for the fix!
